### PR TITLE
Initial Interactor structure

### PR DESCRIPTION
**Pull request by bokimilinkovic**
https://trello.com/c/OgnC00Yi/97-add-book-interactor

### CODE REVIEW

File: interactor/book.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+type store interface {
+	FindBookById(int) (*model.Book, error)
+	CreateBook(*model.Book) error
+	UpdateBook(*model.Book) error
+	FindAllBooks() ([]*model.Book, error)
+	DeleteBook(*model.Book) error
+}
+
+type openLibraryClient interface {
```
Comment (sfqi) [inline interactor/book.go]: openlibraryClient

File: interactor/book.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	FetchBook(isbn string) (*openlibrarydto.Book, error)
+}
+
+type Book struct {
+	Db  store
+	Olc openLibraryClient
```
Comment (sfqi) [inline interactor/book.go]: Unexport both fields
maybe just call them `store` and `openlib`

File: interactor/book.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+func NewBook(store store, olc openLibraryClient) *Book {
+	return &Book{
+		Db:  store,
+		Olc: olc,
+	}
+}
+
+func (bi *Book) FindAll() ([]*model.Book, error) {
```
Comment (sfqi) [inline interactor/book.go]: `b`, not `bi`

File: interactor/book.go (hunk 2 above)
Comment (bokimilinkovic) [reply interactor/book.go]: Do you mean for Types or the name of fields?

Comment (sfqi) [reply interactor/book.go]: fields

**Pull request merged**